### PR TITLE
Added summary on warnings and errors (#2581)

### PR DESCRIPTION
**Pull request by bladerunner2020**
Added a summary message about warnings and errors if any (see #2581) in the following functions:
- generateDocs
- generateJson

Added tests.

### CODE REVIEW

File: src/lib/application.ts [hunk 1]
```diff
@@ -525,6 +525,14 @@ export class Application extends ChildableComponent<
         const start = Date.now();
         out = Path.resolve(out);
         await this.renderer.render(project, out);
+
+        // Generate a message summarizing the number of errors and warnings if there are any
```
Comment (Gerrit0) [inline src/lib/application.ts:529]: I think this is the wrong place to put this. TypeDoc has a `--watch` option, which will result in `generateDocs` and `generateJson` being called many times. If a user has both options turned on, then with this implementation they may get two warnings/errors for each compilation.

I think the way I'd prefer to handle this is to add a public `logRunSummary` (better name?) method on `Application` which is called in `cli.ts` in the appropriate places

Comment (Gerrit0) [reply src/lib/application.ts:529]: This also has the benefit of making the tests for this feature much less hacky - I've very intentionally avoided mocking pieces of TypeDoc in tests (almost) everywhere so far, they tend to lead to testing implementation details+easily broken tests

**Review by Gerrit0 — CHANGES_REQUESTED** [review]
Thanks for the PR! I have a couple concerns

File: src/lib/application.ts (hunk 1 above)
Comment (bladerunner2020) [reply src/lib/application.ts:529]: > I think this is the wrong place to put this. TypeDoc has a `--watch` option, which will result in `generateDocs` and `generateJson` being called many times. If a user has both options turned on, then with this implementation they may get two warnings/errors for each compilation.
> 
> I think the way I'd prefer to handle this is to add a public `logRunSummary` (better name?) method on `Application` which is called in `cli.ts` in the appropriate places

That seems reasonable. I'll take a look at it.

**Comment by bladerunner2020** [comment]
I moved `logRunSummary` to `cli.ts` and agree that it's a better place for it. I also removed the test file as the tests are no longer necessary due to the simplicity of this solution. However, I did not add `logRunSummary` to the Application; I left it as a function in `cli.ts`.

**Review by Gerrit0 — APPROVED** [review]

**Comment by Gerrit0** [comment]
Thanks!

**Pull request merged**